tests/std/tests/P0896R4_views_take/test.cpp

```diff
@@ -0,0 +1,530 @@
+    test::CanCompare{derived_from<Category, forward_iterator_tag> || IsCommon == test::Common::yes},
+    test::ProxyRef{!derived_from<Category, contiguous_iterator_tag>}, test::CanView::yes, test::Copyability::move_only>;
+
+constexpr void move_only_test() {
+    using test::Common, test::Sized;
+    using input      = input_iterator_tag;
+    using fwd        = forward_iterator_tag;
+    using bidi       = bidirectional_iterator_tag;
+    using random     = random_access_iterator_tag;
+    using contiguous = contiguous_iterator_tag;
+
+    test_one(move_only_view<input, Sized::no, Common::no>{some_ints}, only_four_ints);
+    test_one(move_only_view<input, Sized::yes, Common::no>{some_ints}, only_four_ints);
+    test_one(move_only_view<fwd, Sized::no, Common::no>{some_ints}, only_four_ints);
+    test_one(move_only_view<fwd, Sized::yes, Common::no>{some_ints}, only_four_ints);
+    test_one(move_only_view<fwd, Sized::no, Common::yes>{some_ints}, only_four_ints);
+    test_one(move_only_view<fwd, Sized::yes, Common::yes>{some_ints}, only_four_ints);
+    test_one(move_only_view<bidi, Sized::no, Common::no>{some_ints}, only_four_ints);
+    test_one(move_only_view<bidi, Sized::yes, Common::no>{some_ints}, only_four_ints);
+    test_one(move_only_view<bidi, Sized::no, Common::yes>{some_ints}, only_four_ints);
+    test_one(move_only_view<bidi, Sized::yes, Common::yes>{some_ints}, only_four_ints);
+    test_one(move_only_view<random, Sized::no, Common::no>{some_ints}, only_four_ints);
+    test_one(move_only_view<random, Sized::yes, Common::no>{some_ints}, only_four_ints);
+    test_one(move_only_view<random, Sized::no, Common::yes>{some_ints}, only_four_ints);
+    test_one(move_only_view<random, Sized::yes, Common::yes>{some_ints}, only_four_ints);
+    test_one(move_only_view<contiguous, Sized::no, Common::no>{some_ints}, only_four_ints);
+    test_one(move_only_view<contiguous, Sized::yes, Common::no>{some_ints}, only_four_ints);
+    test_one(move_only_view<contiguous, Sized::no, Common::yes>{some_ints}, only_four_ints);
+    test_one(move_only_view<contiguous, Sized::yes, Common::yes>{some_ints}, only_four_ints);
+}
+
+constexpr void output_range_test() {
+#if !defined(__clang__) && !defined(__EDG__) // TRANSITION, VSO-938163
+    if (!is_constant_evaluated())
+#endif // TRANSITION, VSO-938163
+    {
+        using R = test::range<output_iterator_tag, int, test::Sized::no, test::CanDifference::no, test::Common::no,
+            test::CanCompare::no, test::ProxyRef::yes, test::CanView::yes, test::Copyability::move_only>;
+        int some_writable_ints[] = {0, 1, 2, 3};
+        STATIC_ASSERT(same_as<decltype(views::take(R{some_writable_ints}, 99999)), ranges::take_view<R>>);
+        // How do I implement "Fill up to n elements in {output range} with {value}"?
+        ranges::fill(R{some_writable_ints} | views::take(99999), 42);
+        assert(ranges::equal(some_writable_ints, initializer_list<int>{42, 42, 42, 42}));

```

Comment:
This is cool! Should there also be test coverage for when the `views::take` ends before the range does?

Reply:
Sure, why not?